### PR TITLE
Redesign Intacct : Mappings

### PR DESCRIPTION
**Pull request by anishfyle**

### CODE REVIEW

File: src/app/integrations/si/main/mapping/employee-mapping/employee-mapping.component.ts [hunk 1]
```diff
@@ -7,9 +20,212 @@ import { Component, OnInit } from '@angular/core';
...
+        this.filteredMappings = this.mappings.concat();
+        this.isInitialSetupComplete = true;
+        this.isLoading = false;
+      }
+    );
+  }
 
   ngOnInit(): void {
+    this.mappingService.getConfiguration().subscribe((response) => {
+      this.isLoading = true;
```
Comment (ashwin1111) [inline src/app/integrations/si/main/mapping/employee-mapping/employee-mapping.component.ts:225]: not needed, make it true by default

File: src/app/integrations/si/main/mapping/generic-mapping/generic-mapping.component.ts [hunk 2]
```diff
@@ -7,9 +15,87 @@ import { Component, OnInit } from '@angular/core';
...
+  // }
+
+  // MappingStateFilter(state: MappingState): void {
+  //   This.isLoading = true;
+  //   This.selectedMappingFilter = state;
+  //   This.currentPage = 1;
+  //   This.limit = 10;
+  //   This.pageNo = 0;
+  //   This.getFilteredMappings();
+  // }
```
Comment (ashwin1111) [inline src/app/integrations/si/main/mapping/generic-mapping/generic-mapping.component.ts:79]: why are these commented out?

File: src/app/integrations/si/main/mapping/generic-mapping/generic-mapping.component.ts [hunk 3]
```diff
@@ -7,9 +15,87 @@ import { Component, OnInit } from '@angular/core';
...
+    forkJoin([
+      // This.mappingService.getMappingStats(this.sourceType),
+      // This.mappingService.getMappings(this.limit, this.pageNo, this.sourceType, MappingState.ALL)
+    ]).subscribe((response) => {
+      // This.mappingState = response[0];
+      // This.mappings = response[1];
+      // This.filteredMappings = this.mappings.results.concat();
+      // This.totalCount = this.mappings.count;
+      // This.getOps();
+      // This.isLoading = false;
```
Comment (ashwin1111) [inline src/app/integrations/si/main/mapping/generic-mapping/generic-mapping.component.ts:93]: why are these commented?

File: src/app/integrations/si/main/mapping/mapping.component.ts [hunk 4]
```diff
@@ -7,9 +11,47 @@ import { Component, OnInit } from '@angular/core';
...
+  ];
+
+  activeModule: MenuItem;
+
+  constructor(
+    private router: Router,
+    private mappingService: SiMappingsService
+  ) { }
+
+  private snakeToTitleCase(str: string): string {
```
Comment (ashwin1111) [inline src/app/integrations/si/main/mapping/mapping.component.ts:28]: make it a pipe, if it already exist reuse it

File: src/app/integrations/si/main/mapping/mapping.component.ts [hunk 5]
```diff
@@ -7,9 +11,47 @@ import { Component, OnInit } from '@angular/core';
...
+      .replace(/\w\S*/g, (txt) => {
+        return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase(); // Title case
+      });
+  }
+
+  private setupPages(): void {
+    this.isLoading = true;
+    this.mappingService.getMappingSettings().subscribe((response) => {
+      if (response.results && Array.isArray(response.results)) {
+        response.results.forEach((item) => {
```
Comment (ashwin1111) [inline src/app/integrations/si/main/mapping/mapping.component.ts:40]: how does this ensure employee and category comes first in the tabs?

File: src/app/integrations/si/main/mapping/category-mapping/category-mapping.component.ts [hunk 6]
```diff
@@ -7,9 +18,203 @@ import { Component, OnInit } from '@angular/core';
...
+    private mappingService: SiMappingsService,
+    private route: ActivatedRoute,
+    private workspaceService: SiWorkspaceService,
+    private toastService: IntegrationsToastService
+  ) { }
+
+  tableDropdownWidth() {
+    const element = document.querySelector('.p-dropdown-panel.p-component.ng-star-inserted') as HTMLElement;
+    if (element) {
+      element.style.width = '300px';
```
[the remark this answers is not in the document]
Comment (anishfyle) [reply src/app/integrations/si/main/mapping/category-mapping/category-mapping.component.ts:75]: To check this with @DhaaraniCIT

File: src/app/integrations/si/main/mapping/employee-mapping/employee-mapping.component.ts [hunk 7]
```diff
@@ -7,9 +20,212 @@ import { Component, OnInit } from '@angular/core';
...
+  ) { }
+
+  tableDropdownWidth() {
+    const element = document.querySelector('.p-dropdown-panel.p-component.ng-star-inserted') as HTMLElement;
+    if (element) {
+      element.style.width = '300px';
+    }
+  }
+
+  filterOptions() {
```
[the remark this answers is not in the document]
Comment (anishfyle) [reply src/app/integrations/si/main/mapping/employee-mapping/employee-mapping.component.ts:83]: removed this for meanWhile, will address it at last

**Pull request merged**